lib/http-bridge/src/Cardano/Wallet/Network/HttpBridge/Api.hs

```diff
@@ -112,17 +120,21 @@ newtype NetworkName = NetworkName
 instance ToHttpApiData NetworkName where
     toUrlPiece = getNetworkName
 
-instance ToJSON (ApiT SignedTx) where
-    toJSON (ApiT (SignedTx bs))= object ["signedTx" .= c bs]
+instance ToJSON (ApiT (Tx, [TxWitness])) where
+    toJSON (ApiT (tx, wit))= object ["signedTx" .= base64 bytes]
       where
-        c :: ByteString -> String
-        c = B8.unpack . convertToBase Base64
+        bytes :: ByteString
+        bytes = toByteString $ encodeSignedTx (tx, wit)
+        base64 :: ByteString -> String
+        base64 = B8.unpack . convertToBase Base64
 
-instance FromJSON (ApiT SignedTx) where
+instance FromJSON (ApiT (Tx, [TxWitness])) where
     parseJSON = withObject "SignedTx" $ \p -> do
-        base64 <- p .: "signedTx"
-        bs <- either fail return $ c base64
-        return $ ApiT . SignedTx $ bs
+        bs <- (base64 <$> (p .: "signedTx"))
+            >>= either fail return
+        tx <- pure (CBOR.deserialiseFromBytes decodeSignedTx (BL.fromStrict bs))
```

Comment:
is this change compatible with "old data" - if tx witness is stored on a blockchain

Reply:
aha - in fact I see it didn't change at all. You have just moved some code bits here

Reply:
Indeed. The code is the same, but now done within the `http-bridge` package (rather than in the core), because it uses some cbor-specific encoding. So, before that, the core was sending a 'SignedTx ByteString`, where the `ByteString` was supposedly the serialized tx, and it now sends a `(Tx, [TxWitness])` and the network layer takes care of serializing it. This keeps the network layer agnostic to the binary format.